rust/telemetry/src/posthog.rs

```diff
@@ -0,0 +1,251 @@
+use crate::client::TelemetryClient;
+use crate::events::ProductTelemetryEvent;
+use async_trait::async_trait;
+use posthog_rs::{Client, Event};
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tracing::warn;
+
+const POSTHOG_API_KEY: &str = "phc_YeUxaojbKk5KPi8hNlx1bBKHzuZ4FDtl67kH1blv8Bh";
+const UNKNOWN_USER_ID: &str = "UNKNOWN";
+
+pub struct Posthog {
+    client: Client,
```

Comment:
This client doesn't implement batching 

https://github.com/PostHog/posthog-rs/blob/5e2db3ac5c4bcc861d0b02952a2c59a79d384389/src/client/async_client.rs#L29C1-L37C1

We need batching for performance here, and to move the requests out of the critical path

Reply:
The posthog client in python internally batches

Reply:
https://posthog.com/docs/libraries/python